core/agent_core.go

```diff
@@ -146,23 +158,27 @@ func (agent *Agent) initPlugins() error {
 			return fmt.Errorf(logErrorFmt, plug.PluginName, err)
 		}
 		agent.Info(fmt.Sprintf(logSuccessFmt, plug.PluginName))
+		agent.loadedPlugins = append(agent.loadedPlugins, plug)
 	}
 	return nil
 }
 
 // handleAfterInit calls the AfterInit handlers for plugins that can only
 // finish their initialization after  all other plugins have been initialized.
 func (agent *Agent) handleAfterInit() error {
+	agent.loadedPlugins = []*NamedPlugin{} // reset loaded plugins
 	for _, plug := range agent.plugins {
+		// set currently after-initialized plugin name
+		agent.handledPlugin = string(plug.PluginName + " AfterInit()")
 		if plug2, ok := plug.Plugin.(PostInit); ok {
 			agent.Debug("afterInit begin for ", plug.PluginName)
 			err := plug2.AfterInit()
 			if err != nil {
 				agent.Stop()
-
 				return fmt.Errorf(logPostErrorFmt, plug.PluginName, err)
 			}
 			agent.Info(fmt.Sprintf(logPostSuccessFmt, plug.PluginName))
+			agent.loadedPlugins = append(agent.loadedPlugins, plug)
```

Comment:
this needs to be a different collection (please figure out also proper name for init() collection myabe just suffix init)